astropy/table/table.py

```diff
@@ -2345,7 +2345,9 @@ def _replace_column_warnings(self, name, col):
         refcount = None
         old_col = None
 
-        if 'refcount' in warns and name in self.colnames:
+        # sys.getrefcount is CPython specific and not on PyPy.
+        if ('refcount' in warns and name in self.colnames and
```

Comment:
you can `from numpy.testing import HAS_REFCOUNT` and then `pytest.mark.skip` based on it -- this is done in numpy too; e.g., https://github.com/numpy/numpy/blob/fbe1b65f5f66ca5a416b23194577a1ed5a4cf1bd/numpy/core/tests/test_ufunc.py#L2498-L2527

Definition at https://github.com/numpy/numpy/blob/fbe1b65f5f66ca5a416b23194577a1ed5a4cf1bd/numpy/core/tests/test_ufunc.py#L2498-L2527

Reply:
`HAS_REFCOUNTS` is used to skip tests. Maybe that would be a better approach: skip tests that hit this code path if the interpreter does not support `HAS_REFCOUNT`. I would be wary of merging this as-is, it seems it would lead to resource leaks, which will turn up as other random errors (unable to open files or mmaps).

Reply:
The original problem that prompted this over at conda-forge was not due to testing. It was because PySynphot read in a Vega spectrum to set up Vega magnitude zero point on import. That kicked off the whole `astropy.io.fits` memmap count thingy because memmap is True by default.

Reply:
Just to be clear: I don't suggest to leave it at this, but just to split this into ensuring `PyPy` can run at all, and ensuring that it always runs correctly (and for as long as needed, without memory leaks).

Reply:
@mhvk , I don't feel comfortable using `from numpy.testing import HAS_REFCOUNT` in non-test code.

Reply:
Again, I want to stress that the original error was not from a test. It was from pysynphot opening a FITS file on import with default memmap setting (True).

Reply:
@pllim - yes, that makes sense. Happy to go with what you have here; I think the main thing is to get something that doesn't fail -- we can worry about fixing leaks later.

Reply:
OK I rebased and took this out of draft...